build_tools/azure/install.sh

```diff
@@ -33,155 +30,158 @@ setup_ccache() {
     ccache -M 256M
 }
 
-# defines the get_dep and show_installed_libraries functions
-source build_tools/shared.sh
+pre_python_environment_install() {
+    if [[ "$DISTRIB" == "ubuntu" ]]; then
+        sudo add-apt-repository --remove ppa:ubuntu-toolchain-r/test
+        sudo apt-get update
+        sudo apt-get install python3-scipy python3-matplotlib\
+             libatlas3-base libatlas-base-dev python3-virtualenv ccache
+
+    elif [[ "$DISTRIB" == "debian-32" ]]; then
+        apt-get update
+        apt-get install -y python3-dev python3-numpy python3-scipy \
+                python3-matplotlib libatlas3-base libatlas-base-dev \
+                python3-virtualenv python3-pandas ccache
+    elif [[ "$DISTRIB" == "conda-mamba-pypy3" ]]; then
+        # condaforge/mambaforge-pypy3 needs compilers
+        apt-get -yq update
+        apt-get -yq install build-essential
+
+    elif [[ "$BUILD_WITH_ICC" == "true" ]]; then
+        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+        rm GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+        sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
+        sudo apt-get update
+        sudo apt-get install intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic
+        source /opt/intel/oneapi/setvars.sh
 
-if [[ "$DISTRIB" == "conda" || "$DISTRIB" == *"mamba"* ]]; then
+        # The "build_clib" command is implicitly used to build "libsvm-skl".
+        # To compile with a different compiler, we also need to specify the
+        # compiler for this command
+        python setup.py build_ext --compiler=intelem -i build_clib --compiler=intelem
```

Comment:
I think we still have to wait till `scikit_learn_install` to run `build_ext` so we use the correct python environment?

Currently on `main`, the conda environment is created first and `setup.py build_ext` is called in that enviornment.

Reply:
Good catch I fixed this and use `[icc-build]` in my last commit to make sure to test the BUILD_WITH_ICC logic and the CI passes (here is the [log](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=38396&view=logs&jobId=8628a494-79d0-53fa-274c-1b00464f7121&j=8628a494-79d0-53fa-274c-1b00464f7121))